leader-election-impl/src/main/java/com/palantir/leader/LeaderElectionServiceBuilder.java

```diff
@@ -0,0 +1,145 @@
+        return this;
+    }
+
+    public LeaderElectionServiceBuilder eventRecorder(PaxosLeaderElectionEventRecorder eventRecorder) {
+        this.eventRecorder = Preconditions.checkNotNull(eventRecorder, "eventRecorder cannot be null");
+        return this;
+    }
+
+    public LeaderElectionServiceBuilder pingRate(Duration pingRate) {
+        this.pingRate = Preconditions.checkNotNull(pingRate, "pingRate cannot be null");
+        return this;
+    }
+
+    public LeaderElectionServiceBuilder randomWaitBeforeProposingLeadership(
+            Duration randomWaitBeforeProposingLeadership) {
+        this.randomWaitBeforeProposingLeadership = Preconditions.checkNotNull(
+                randomWaitBeforeProposingLeadership, "randomWaitBeforeProposingLeadership cannot be null");
+        return this;
+    }
+
+    public LeaderElectionServiceBuilder leaderUuid(UUID leaderUuid) {
+        this.leaderUuid = Preconditions.checkNotNull(leaderUuid, "leaderUuid cannot be null");
+        return this;
+    }
+
+    public LeaderElectionServiceBuilder quorumSize(int quorumSize) {
+        this.quorumSize = Preconditions.checkNotNull(quorumSize, "quorumSize cannot be null");
+        return this;
+    }
+
+    public LeaderElectionService build() {
+        return new PaxosLeaderElectionService(
+                buildProposer(),
+                knowledge(),
+                leaderPinger(),
+                acceptorClient(),
+                learnerClient(),
+                pingRate().toMillis(),
+                randomWaitBeforeProposingLeadership().toMillis(),
+                eventRecorder());
+    }
+
+    private PaxosProposer buildProposer() {
+        return PaxosProposerImpl.newProposer(acceptorClient(), learnerClient(), quorumSize(), leaderUuid());
+    }
+
+    private int quorumSize() {
+        Preconditions.checkArgument(quorumSize > 0, "quorumSize not set");
+        return quorumSize;
+    }
+
+    private PaxosAcceptorNetworkClient acceptorClient() {
+        return Preconditions.checkNotNull(acceptorClient, "acceptorClient not set");
+    }
+
+    private PaxosLearnerNetworkClient learnerClient() {
+        return Preconditions.checkNotNull(learnerClient, "learnerClient not set");
+    }
+
+    private PaxosLearner knowledge() {
+        return Preconditions.checkNotNull(knowledge, "knowledge not set");
+    }
+
+    private LeaderPinger leaderPinger() {
+        return Preconditions.checkNotNull(leaderPinger, "leaderPinger not set");
+    }
+
+    private PaxosLeaderElectionEventRecorder eventRecorder() {
+        return Preconditions.checkNotNull(eventRecorder, "eventRecorder not set");
+    }
+
+    private Duration pingRate() {
+        return Preconditions.checkNotNull(pingRate, "pingRate not set");
+    }
+
+    private Duration randomWaitBeforeProposingLeadership() {
+        return Preconditions.checkNotNull(randomWaitBeforeProposingLeadership,
+                "randomWaitBeforeProposingLeadership not set");
+    }
```

Comment:
nit / maybe non-actionable: Do we want to check, along the lines of the quorum thing, that durations are not negative?